src/emu/sound.cpp

```diff
@@ -859,8 +866,13 @@ void sound_stream::postload()
 {
 	// set the end time of all of our streams to now
 	for (auto &output : m_output)
-		output.set_end_time(m_device.machine().time());
-
+	{
+		// previous : machine time
+		// output.set_end_time(m_device.machine().time());
+		// printf("postload %s %f\n", name().c_str(), m_last_update_end.as_double());
+		// resampler is zero
+		output.set_end_time(m_last_update_end > attotime::zero ? m_last_update_end : m_device.machine().time());
+	}
```

Comment:
I don't understand.  output has a end time stored somewhere in some form, why don't you save that?

Reply:
I would need a fixed array to cache the value of `output.end_time()`. And in addition a `pre_save` handler. The end time for all outputs is always the same. That's why I used this approach.
In addition there is stuff like (line 680)
```
	// ignore any update requests if we're already up to date
	attotime start = m_output[0].end_time();
	attotime end = m_device.machine().time();
	if (start >= end)
	{
		m_last_update_end = start;
		return;
	}

	// regular update then
	update_view(start, end);
```
This indicates that the correct sound device time is end_time() for `start >= end`. In the other case it is `machine().time()`.